openpype/hosts/max/plugins/publish/extract_review_animation.py

```diff
@@ -0,0 +1,135 @@
+        self.log.debug(f"{representation}")
+
+        if "representations" not in instance.data:
+            instance.data["representations"] = []
+        instance.data["representations"].append(representation)
+
+    def get_files(self, filename, start, end, ext):
+        file_list = []
+        for frame in range(int(start), int(end) + 1):
+            actual_name = "{}.{:04}.{}".format(
+                filename, frame, ext)
+            file_list.append(actual_name)
+
+        return file_list
+
+    def set_preview_arg(self, instance, filepath,
+                        start, end, fps):
+        job_args = list()
+        default_option = f'CreatePreview filename:"{filepath}"'
+        job_args.append(default_option)
+
+        frame_option = f"outputAVI:false start:{start} end:{end} fps:{fps}" # noqa
+        job_args.append(frame_option)
+        rndLevel = instance.data.get("rndLevel")
+        if rndLevel:
+            option = f"rndLevel:#{rndLevel}"
+            job_args.append(option)
+        percentSize = instance.data.get("percentSize")
+        if percentSize:
+            size = int(percentSize)
+            option = f"percentSize:{size}"
+            job_args.append(option)
+        dspGeometry = instance.data.get("dspGeometry")
+        if dspGeometry:
+            option = f"dspGeometry:{dspGeometry}"
+            job_args.append(option)
+        dspShapes = instance.data.get("dspShapes")
+        if dspShapes:
+            option = f"dspShapes:{dspShapes}"
+            job_args.append(option)
+        dspLights = instance.data.get("dspLights")
+        if dspLights:
+            option = f"dspShapes:{dspLights}"
+            job_args.append(option)
+        dspCameras = instance.data.get("dspCameras")
+        if dspCameras:
+            option = f"dspCameras:{dspCameras}"
+            job_args.append(option)
+        dspHelpers = instance.data.get("dspHelpers")
+        if dspHelpers:
+            option = f"dspHelpers:{dspHelpers}"
+            job_args.append(option)
+        dspParticles = instance.data.get("dspParticles")
+        if dspParticles:
+            option = f"dspParticles:{dspParticles}"
+            job_args.append(option)
+        dspBones = instance.data.get("dspBones")
+        if dspBones:
+            option = f"dspBones:{dspBones}"
+            job_args.append(option)
+        dspBkg = instance.data.get("dspBkg")
+        if dspBkg:
+            option = f"dspBkg:{dspBkg}"
+            job_args.append(option)
+        dspGrid = instance.data.get("dspGrid")
+        if dspGrid:
+            option = f"dspBkg:{dspBkg}"
+            job_args.append(option)
+        dspSafeFrame = instance.data.get("dspSafeFrame")
+        if dspSafeFrame:
+            option = f"dspSafeFrame:{dspSafeFrame}"
+            job_args.append(option)
+        dspFrameNums = instance.data.get("dspFrameNums")
+        if dspFrameNums:
+            option = f"dspFrameNums:{dspFrameNums}"
+            job_args.append(option)
```

Comment:
I'd say this is probably easier to maintain when written like this:

```
options = [
    "dspSafeFrame", "dspFrameNums", "dskBkg", # etc...
] 
for key in options:
    if key not in instance.data:
        continue
    value = instance.data[key]
    job_args.append(f"{key}:{value}")
```

Especially since there's already human error in the code currently, the `dspGrid` branch doesn't set the right value but writes ` dspBkg` instead.


Only `rndLevel` seems to be the odd one out since it adds a `#` in front of the value?

Also note that your current logic would never pass the value if it was ever e.g. 0 since it wouldn't pass the if statement. My replacement logic would still pass it, so make sure that's what you want. I suspect that's not what you want since they seem to each be Boolean values and I'm not sure if then you'd want to pass `False` explicitly?

If not, you could do:
```python
options = [
    "dspSafeFrame", "dspFrameNums", "dskBkg", # etc...
] 
for key in options:
    enabled= instance.data.get(key)
    if enabled:
        job_args.append(f"{key}:{enabled}")
```

Reply:
Also, the logic appears to be pretty similar to the other plugin in this PR so I'd try and see where you could potentially deduplicate the code too and re-use the same logic.